src/shortcodes/gist.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const {extractLayoutString} = require('../helpers/shortcodeLayoutParser');
+
+module.exports = (...args) => {
+  if (args.length == 0) {
+    throw new Error('Missing gist id');
+  }
+  const id = args[0];
+  const layout = extractLayoutString(args.splice(1));
```

Comment:
does this fail if the layout isn't provided (since the second `defaultLayout` arg isn't supplied)? If it does, is the failure clear? 

EDIT: see you are testing for this case in the below test (yay!), but I am not clear that the error stack would be obviously the result of a required arg

Reply:
I've rewritten the exception to a more explicit one mentioning that a `height` should be provided.